packages/protocol/contracts/identity/Escrow.sol

```diff
@@ -168,17 +250,54 @@ contract Escrow is
     EscrowedPayment memory payment = escrowedPayments[paymentId];
     require(payment.token != address(0) && payment.value > 0, "Invalid withdraw value.");
 
+    bool passedCheck = false;
```

Comment:
I broke it into a couple of functions to hopefully help readability (will merge but please take a look in the big PR and lmk if it's more readable!); unfortunately I think we can't fully get rid of the passedCheck, in particular the if/else on trustedIssuers.length > 0, since we want v1Attestations to be checked only when specified as a trustedIssuer + for backwards compatibility (i.e. trustedIssuers not set), but not as a default/backup in case other attestations are not found. Maybe I'm misunderstanding you though, lmk if you meant something else 🙏